Reference/Searching/Examine/Quick-Start/index.md

```diff
@@ -66,208 +103,162 @@ Umbraco ships with three indexes:
-                searchTerm = string.IsNullOrEmpty(Context.Request.Query["query"])
-                    ? string.Empty
-                    : Context.Request.Query["query"];
-                if (searchTerm == string.Empty)
-                {
-                    <p>Enter search term</p>
-                }
-                else
-                {
-                    //perform the search
-                    //first we try to get the index, it is the ExternalIndex as we don't want to return unpublished things
-                    //it returns the index in the var index
-                    //be sure to add "@using Umbraco.Examine;" at the top of the view
-                    if(ExamineManager.TryGetIndex("ExternalIndex", out var index))
-                    {
-                        var searcher = index.Searcher;
-                        var results = searcher.CreateQuery("content").NodeTypeAlias("person").And().Field("nodeName", searchTerm).Execute();
-                        if (results.Any())
-                        {
-                            <ul>
-                                @foreach (var result in results)
-                                {
-                                    if (result.Id != null)
-                                    {
-                                        var node = Umbraco.Content(result.Id);
-                                        <li>
-                                            <a href="@node.Url()">@node.Name</a>
-                                        </li>
-                                    }
-                                }
-                            </ul>
-                        }
-                        else
-                        {
-                            <p>No results found for query @searchTerm</p>
-                        }
-                    }
-                    return;
-                }
-            }
-        </div>
-        <div class="employee-grid">
-            @foreach (Person person in Model.Children<Person>())
-            {
+## Using the service and view model in the controller
+Now that we've created our service that handles the actual search logic and ViewModel, lets look at using it in the controller.
+We will want to update the `Index()` method to get out the queryString from the request, then create a viewmodel and populate the `SearchResults` property by using our service
+```csharp
+public override IActionResult Index()
+{
+    // Get the queryString from the request
+    string queryString = _httpContextAccessor.HttpContext?.Request.Query["query"] ?? string.Empty;
+    // Create the viewModel and pass it to the view
+    SearchViewModel viewModel = new(CurrentPage!, new PublishedValueFallback(_serviceContext, _variationContextAccessor))
+    {
+        SearchResults = _searchService.SearchContentNames(queryString),
+        HasSearched = !string.IsNullOrEmpty(queryString),
+    };
+
+    return CurrentTemplate(viewModel);
+}
+```
```

Comment:
This code sample does not work on its own - it references services that aren't available from the code samples before. Suggest replacing it with the following (including required namespaces for V9).

```suggestion
```csharp
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.ViewEngines;
using Microsoft.Extensions.Logging;
using MyStarterKitSite.Models;
using MyStarterKitSite.Services;
using Umbraco.Cms.Core.Models.PublishedContent;
using Umbraco.Cms.Core.Web;
using Umbraco.Cms.Web.Common.Controllers;

namespace MyStarterKitSite.Controllers;

public class PeopleController : RenderController
{
    private readonly IPublishedValueFallback _publishedValueFallback;
    private readonly ISearchService _searchService;
    
    public PeopleController(
        ILogger<RenderController> logger,
        ICompositeViewEngine compositeViewEngine,
        IUmbracoContextAccessor umbracoContextAccessor, 
        IPublishedValueFallback publishedValueFallback, 
        ISearchService searchService)
        : base(logger,
            compositeViewEngine,
            umbracoContextAccessor)
    {
        _publishedValueFallback = publishedValueFallback;
        _searchService = searchService;
    }

    public override IActionResult Index()
    {
        // Get the queryString from the request
        string queryString = HttpContext.Request.Query["query"];
        
        // Create the view model and pass it to the view
        SearchViewModel viewModel = new(CurrentPage!, _publishedValueFallback)
        {
            SearchResults = _searchService.SearchContentNames(queryString),
            HasSearched = !string.IsNullOrEmpty(queryString),
        };

        return CurrentTemplate(viewModel);
    }
}
```
```